### PR TITLE
Desing Patterns by Deyan Sadinov

### PR DESCRIPTION
**Pull request by deyansadinov**
TASK 1

### CODE REVIEW

File: src/com/clouway/task1/factorypattern/abstractfactory/AbstractFactoryPatternDemo.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.clouway.task1.factorypattern.abstractfactory;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class AbstractFactoryPatternDemo {
+  public static void main(String[] args) {
+
+    //get shape factory
+    AbstractFactory shapeFactory = FactoryProducer.getFactory("SHAPE");
```
Comment (mgenov) [inline src/com/clouway/task1/factorypattern/abstractfactory/AbstractFactoryPatternDemo.java]: The abstract factory is creating factories. So I think that FactoryProducer is not a good name in the context of this pattern.

File: src/com/clouway/task1/factorypattern/reflection/HousePartsFactory.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.clouway.task1.factorypattern.reflection;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class HousePartsFactory {
+
+  public HouseParts getHouseParts(String type) {
+    if (type.equals("window")) {
```
Comment (mgenov) [inline src/com/clouway/task1/factorypattern/reflection/HousePartsFactory.java]: I would like to see better approach.

File: src/com/clouway/task1/factorypattern/abstractfactory/AbstractFactoryPatternDemo.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.clouway.task1.factorypattern.abstractfactory;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class AbstractFactoryPatternDemo {
+  public static void main(String[] args) {
+
+    //get shape factory
+    AbstractFactory shapeFactory = Producer.getFactory("SHAPE");
```
Comment (mgenov) [inline src/com/clouway/task1/factorypattern/abstractfactory/AbstractFactoryPatternDemo.java]: Already saw Producer used in another PR.

File: src/com/clouway/task1/factorypattern/abstractfactory/Producer.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.clouway.task1.factorypattern.abstractfactory;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class Producer {
+  public static AbstractFactory getFactory(String choice){
+    if (choice.equalsIgnoreCase("SHAPE")){
```
Comment (mgenov) [inline src/com/clouway/task1/factorypattern/abstractfactory/Producer.java]: "SHAPE".equalsIgnoreCase(choice);  prevents NPE when the provided parameter is null.

File: src/com/clouway/task1/factorypattern/abstractfactory/ColorFactory.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package com.clouway.task1.factorypattern.abstractfactory;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class ColorFactory extends AbstractFactory {
+  @Override
+  Shape getShape(String shape) {
```
Comment (mgenov) [inline src/com/clouway/task1/factorypattern/abstractfactory/ColorFactory.java]: Seems like bad example, cause your ColorFactory is producing shapes and also it returns null. 

I would like to see better example for the whole AbstractFactory.

Comment (deyansadinov) [inline src/com/clouway/task1/factorypattern/abstractfactory/ColorFactory.java]: I wrote a new abstract factory

**Comment by mgenov** [comment]
Still not an abstract factory.

**Pull request merged**